tests/Pingalot.Core.Tests/PingSessionTests.cs

```diff
@@ -32,11 +32,86 @@ public void TestPingStatisticsDivideByZero()
 				RequestTime = DateTime.Now
 			};
 
-			pingRequests.Add(pingRequest);
+			var testPingSession = new PingSession(startTime);
 
-			var testPingSession = new PingSession(startTime, endTime, duration, pingRequests);
+			testPingSession.AddSinglePingResult(duration, pingRequest);
 
+			// When we have a single failed ping request - our AverageRoundtrip should be 0
 			Assert.AreEqual(testPingSession.AverageRoundtrip, 0, "If PacketsReceived is 0, when we attempt to calculate AverageRoundTrip we may trigger a divide by zero exception.");
 		}
+
+		[TestMethod]
+		public void ConfirmPingSessionStatistics()
+		{
+			// Mock up PingRequests with specific values
+			// Manually calculate the expected statistics 
+			// Assert stats are correct
+
+			var startTime = new DateTime(2022, 5, 19, 14, 30, 0);
+			TimeSpan duration = new System.TimeSpan(0, 0, 0, 30);
+			var endTime = startTime.Add(duration);
+
+			var testPingSession = new PingSession(startTime);
+
+			// Create 6 successful pings 10ms RT
+			for (int i = 0; i < 6; i++)
+			{
+				var pingRequest = new PingRequest
+				{
+					Address = IPAddress.Loopback,
+					Status = IPStatus.Success,
+					RoundtripTime = 10,
+					TimeToLive = 58,
+					BufferLength = 32,
+					HasMatchingBuffer = true,
+					RequestTime = DateTime.Now
+				};
+
+				testPingSession.AddSinglePingResult(duration, pingRequest);
+			}
+
+			// Create 2 successful pings 5ms RT 
+			for (int i = 0; i < 3; i++)
+			{
+				var pingRequest = new PingRequest
+				{
+					Address = IPAddress.Loopback,
+					Status = IPStatus.Success,
+					RoundtripTime = 5,
+					TimeToLive = 58,
+					BufferLength = 32,
+					HasMatchingBuffer = true,
+					RequestTime = DateTime.Now
+				};
+
+				testPingSession.AddSinglePingResult(duration, pingRequest);
+			}
+
+			// Create 2 TimedOut pings
+			for (int i = 0; i < 2; i++)
+			{
+				var pingRequest = new PingRequest
+				{
+					Address = IPAddress.Loopback,
+					Status = IPStatus.TimedOut,
+					RoundtripTime = 0,
+					TimeToLive = 0,
+					BufferLength = 0,
+					HasMatchingBuffer = false,
+					RequestTime = DateTime.Now
+				};
+
+				testPingSession.AddSinglePingResult(duration, pingRequest);
+			}
```

Comment:
Rather than do these loops, I'd rather create a private static function in the test class that can create the `PingRequest` for us. Something that takes everything _but_ the address (we can default to `IPAddress.Loopback` unless we want to test IPv6 but that is kinda outside of scope for these tests).

Something like: `private static PingRequest CreatePingRequest(IPStatus status, double roundtripTime = 0, ...)` etc.

Then you'd just call this a few times in a row:
```csharp
testPingSession.AddSinglePingResult(duration, CreatePingResult(/* the parameters */));
testPingSession.AddSinglePingResult(duration, CreatePingResult(/* the parameters */));
...
```

I'd still avoid passing in something like `DateTime.Now` as it creates variability in test results (though depends on what we're testing).